pkg/scale/decode.go

```diff
@@ -160,7 +155,7 @@ func (ds *decodeState) unmarshal(dstv reflect.Value) (err error) {
 		case reflect.Slice:
 			err = ds.decodeSlice(dstv)
 		default:
-			err = fmt.Errorf("unsupported type: %T", in)
+			err = fmt.Errorf("invalid type: %T", in)
```

Comment:
unsupported is more accurate that invalid. I would suggest to change this back. You could use `ErrUnsupportedType`